src/device.hpp

```diff
@@ -217,13 +227,37 @@ struct cvk_device : public _cl_device_id,
         return size * percentage_of_available_memory_reported;
     }
 
-    uint64_t memory_size() const {
+    uint64_t global_memory_size() const {
         return std::min(max_alloc_size() * 4, actual_memory_size());
     }
 
     uint64_t max_alloc_size() const {
-        uint64_t max_buffer_size = m_properties.limits.maxStorageBufferRange;
-        return std::min(max_buffer_size, actual_memory_size());
+        // Min memory as per the specs.
+        auto specMinAllocSz =
+            std::max(std::min((uint64_t)(1024 * 1024 * 1024),
+                              (uint64_t)(actual_memory_size())),
+                     (uint64_t)(32 * 1024 * 1024));
+
+        // Max memory allocation for single buffer is set in config.def to 1024
+        // mb - minimum required by spec. For single allocation this value can
+        // be adjusted with environment variable CLVK_MAX_ALLOC_SIZE_MB For
+        // multiple allocations(total memory allocations), environment var
+        // CLVK_PERCENTAGE_OF_AVAILABLE_MEMORY_REPORTED can be adjusted.
+        auto maxAllocSz =
+            std::min(m_mtn3Properties.maxMemoryAllocationSize,
+                     (uint64_t)config.max_alloc_size_mb() * 1024 * 1024);
```

Comment:
I'd rather we set the default value of the option to something that's obviously wrong like zero and only apply the min if the option is set (you can test `config.max_alloc_size_mb.set`). The spec does not really say 1024MB as clearly...